scripts/testing_scripts/test_all_sandia

```diff
@@ -387,7 +387,8 @@ elif [ "$MACHINE" = "apollo" ]; then
 
   SKIP_HWLOC=True
 
-  BASE_MODULE_LIST="sems-env,kokkos-env,kokkos-hwloc/1.10.1/base,sems-<COMPILER_NAME>/<COMPILER_VERSION>"
+  GCC_MODULE_LIST="sems-env,kokkos-env,kokkos-hwloc/1.10.1/base,sems-<COMPILER_NAME>/<COMPILER_VERSION>"
+  NONGCC_MODULE_LIST="sems-env,kokkos-env,sems-gcc/5.3.0,sems-<COMPILER_NAME>/<COMPILER_VERSION>,kokkos-hwloc/1.10.1/base"
```

Comment:
@crtrott why is it necessary to now load `sems-gcc/5.3.0` for the intel builds? This isn't done on other test beds, e.g. bowman.

Reply:
Otherwise the GCC intel is using is the system one GCC 4.4. And the C++11 `std::vector::insert` isn't implemented in it, so my unit test which checks that `std::vector` and `Kokkos::vector` do the same thing doesn't compile.

Reply:
Oh I should maybe explain that standard library for Intel is coming from GCC. I.e. the header files etc. for stuff like std::vector are coming from GCC when you compile with intel.